### PR TITLE
Add `ProjectIsolation` integration test executer

### LINKED ISSUE #25178: Run Gradle test suite with Project Isolation enabled
By introducing a new test executor like we did for `instant execution`.

### PR DESCRIPTION
**Pull request by 6hundreds**
Fixes #25178

### CODE REVIEW

**Comment by 6hundreds** [comment]
@bot-gradle test this

File: build-logic/basics/src/main/kotlin/gradlebuild/basics/testing/TestType.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ import org.gradle.api.tasks.testing.Test
 
 
 enum class TestType(val prefix: String, val executers: List<String>) {
-    INTEGRATION("integ", listOf("embedded", "forking", "noDaemon", "parallel", "configCache")),
+    INTEGRATION("integ", listOf("embedded", "forking", "noDaemon", "parallel", "configCache", "projectIsolation")),
```
Comment (bamboo) [inline build-logic/basics/src/main/kotlin/gradlebuild/basics/testing/TestType.kt:23]: ❌ "isolatedProjects"
```suggestion
    INTEGRATION("integ", listOf("embedded", "forking", "noDaemon", "parallel", "configCache", "isolatedProjects")),
```

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java [hunk 2]
```diff
@@ -39,7 +39,8 @@ private enum Executer {
         forking(true),
         noDaemon(true),
         parallel(true, true),
-        configCache(true);
+        configCache(true),
+        projectIsolation(true);
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java:43]:
```suggestion
        isolatedProjects(true);
```

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java [hunk 3]
```diff
@@ -134,6 +135,8 @@ private GradleExecuter createExecuter(Executer executerType) {
                 return new DaemonGradleExecuter(getDistribution(), getTestDirectoryProvider(), gradleVersion, buildContext);
             case configCache:
                 return new ConfigurationCacheGradleExecuter(getDistribution(), getTestDirectoryProvider(), gradleVersion, buildContext);
+            case projectIsolation:
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java:138]:
```suggestion
            case isolatedProjects:
```

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java [hunk 4]
```diff
@@ -134,6 +135,8 @@ private GradleExecuter createExecuter(Executer executerType) {
...
+                return new ProjectIsolationGradleExecuter(getDistribution(), getTestDirectoryProvider(), gradleVersion, buildContext);
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java:139]:
```suggestion
                return new IsolatedProjectsGradleExecuter(getDistribution(), getTestDirectoryProvider(), gradleVersion, buildContext);
```

**Review by bamboo — CHANGES_REQUESTED** [review]
Let's keep using `IsolatedProjects` for symbols.

**Review by bamboo — APPROVED** [review]
💯

**Comment by 6hundreds** [comment]
@bot-gradle test and merge

**Pull request merged**